cumulusci/robotframework/Salesforce.py

```diff
@@ -488,10 +490,93 @@ def salesforce_insert(self, obj_name, **kwargs):
         """ Inserts a Salesforce object setting fields using kwargs and returns the id """
         self.builtin.log("Inserting {} with values {}".format(obj_name, kwargs))
         obj_class = getattr(self.cumulusci.sf, obj_name)
+
         res = obj_class.create(kwargs)
         self.store_session_record(obj_name, res["id"])
+
         return res["id"]
 
+    def salesforce_init_object(self, obj_name, **fields):
+        obj = {"attributes": {"type": obj_name}}  # Object type to create
+        obj.update(fields)
+        return obj
+
+    def salesforce_init_objects(self, obj_name, number_to_create, **fields):
+        """Create an array of dictionaries with template-formatted arguments appropriate for a Collection Insert.
+            Use ``{number}`` to represent the unique index of the row in the list of rows, ``{random_str}`` to represent a random string
+            and a string that consists of just ``{int}`` to generate an actual integer (as opposed to a string-encoded number)
+
+            For example:
+
+                | @{objects} =  Salesforce Init Objects  Contact  3
+                | ...  FirstName="User {number}"
+                | ...  LastName="{random_str}"
+                | ...  Age="{int()}
+
+            Which would generate:
+
+                | [{'FirstName': '"User 0"', 'LastName': '"u4PNlGUD"', 'type': 'Contact', 'Age': 0},
+                |  {'FirstName': '"User 1"', 'LastName': '"Kyd4PC5x"', 'type': 'Contact', 'Age': 1},
+                |  {'FirstName': '"User 2"', 'LastName': '"n4EGs4Vp"', 'type': 'Contact', 'Age': 2}]
+           """
+        objs = []
+
+        def format_str(value, i):
+            if hasattr(value, "format"):  # Duck-check for if it is string-like
+                if value.strip().lower() == "{int}":
+                    return i
+                else:
+                    return value.format(
+                        number=i, random_str=String().generate_random_string()
+                    )
+            else:
+                return value
+
+        for i in range(int(number_to_create)):
+            formatted_fields = {
+                name: format_str(value, i) for name, value in fields.items()
+            }
+            newobj = self.salesforce_init_object(obj_name, **formatted_fields)
+            objs.append(newobj)
+
+        return objs
+
+    def salesforce_collection_insert(self, objects):
+        """Inserts up to 200 records that were created with Salesforce Init Objects"""
```

Comment:
This says "up to 200 records" but that doesn't seem to be enforced anywhere. If there's a hard limit on the number of objects passed in, that should be checked for and an appropriate error thrown.

Reply:
I think if this 200 is a hard limit, and this is a user-facing keyword, we should probably be prepared to catch errors if we exceed that limit and return a useful error rather than whatever error the server throws.